src/modules/vtol_att_control/tiltrotor.cpp

```diff
@@ -220,45 +216,50 @@ void Tiltrotor::process_mc_data()
 	}
  }
 
-void Tiltrotor::process_fw_data()
-{
-	fill_att_control_output();
-}
-
 void Tiltrotor::update_transition_state()
 {
 	if (_vtol_schedule.flight_mode == TRANSITION_FRONT_P1) {
+		// for the first part of the transition the rear rotors are enabled
+		if (_rear_motors != ENABLED) {
+			set_rear_motor_state(ENABLED);
+		}
 		// tilt rotors forward up to certain angle
-		if (_params_tiltrotor.front_trans_dur <= 0.0f) {
-			_tilt_control = _params_tiltrotor.tilt_transition;
-		} else if (_tilt_control <= _params_tiltrotor.tilt_transition) {
-			_tilt_control = _params_tiltrotor.tilt_mc + fabsf(_params_tiltrotor.tilt_transition - _params_tiltrotor.tilt_mc) * 
-						(float) hrt_elapsed_time(&_vtol_schedule.transition_start) / (_params_tiltrotor.front_trans_dur * 1000000.0f);
+		if (_tilt_control <= _params_tiltrotor.tilt_transition) {
+			_tilt_control = _params_tiltrotor.tilt_mc +  fabsf(_params_tiltrotor.tilt_transition - _params_tiltrotor.tilt_mc)*(float)hrt_elapsed_time(&_vtol_schedule.transition_start)/(_params_tiltrotor.front_trans_dur*1000000.0f);
 		}
 
 		// do blending of mc and fw controls
-		if (_airspeed->true_airspeed_m_s >= ARSP_BLEND_START && _params_tiltrotor.airspeed_trans - ARSP_BLEND_START > 0.0f) {
+		if (_airspeed->true_airspeed_m_s >= ARSP_BLEND_START) {
 			_roll_weight_mc = 1.0f - (_airspeed->true_airspeed_m_s - ARSP_BLEND_START) / (_params_tiltrotor.airspeed_trans - ARSP_BLEND_START);
 		} else {
 			// at low speeds give full weight to mc
 			_roll_weight_mc = 1.0f;
 		}
 
-		_roll_weight_mc = math::constrain(_roll_weight_mc, 0.0f, 1.0f);
+		// disable mc yaw control once the plane has picked up speed
+		_yaw_weight_mc = 1.0f;
+		if (_airspeed->true_airspeed_m_s > 5.0f) {
```

Comment:
I can see we shouldn't add a parameter for this yet till we are sure we need it but at least a constant for now?